src/main/webapp/app/entities/system-notification/system-notification.service.ts

```diff
@@ -50,7 +51,14 @@ export class SystemNotificationService {
         return this.http.delete<any>(`${this.resourceUrl}/${id}`, { observe: 'response' });
     }
 
-    getActiveNotification(): Observable<SystemNotification> {
+    getActiveNotification(): Observable<SystemNotification | null> {
+        // If the user is not authenticated we do an explicit request for an active system notification.
+        if (!this.accountService.isAuthenticated()) {
+            return this.http
+                .get<SystemNotification>(`${this.resourceUrl}/active-notification`, { observe: 'response' })
+                .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)))
+                .pipe(map(res => res.body));
+        }
         return this.notificationService.getRecentSystemNotification();
```

Comment:
Why are there methods related to system notifications in the `notifiation.service` anyway? Since we now want system notifications to be available at all times, it might be better to really separate the "regular" notification service from this system notification service, i.e. only return user specific notifications in the `for-user` endpoint and always fetch all system notifications in this method (for all users).

Reply:
Good point, that would significantly uncomplicate the implementation for the system notifications. I will implement that in a follow-up PR.